spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -335,9 +345,9 @@ private void applySelectWithFilter(Object[] parameters, Builder builder) {
 						"Too few parameters are provided for query method: " + getQueryMethod().getName());
 			}
 
-			Value convertedValue = convertParam(persistentProperty, it.next());
+			Value convertedValue = convertParam(properties.get(properties.size() - 1), it.next());
```

Comment:
Is there possibility of property name clashes if parent class and child class both have a property named, say, `name`?